libraries/security_cmd.rb

```diff
@@ -0,0 +1,44 @@
+include Chef::Mixin::ShellOut
+
+module MacOS
+  class SecurityCommand
+    attr_reader :cert
+    attr_reader :keychain
+    attr_reader :security
+
+    def initialize(cert, keychain)
+      @cert = cert
+      @keychain = keychain
+      @security = '/usr/bin/security'
+    end
+
+    def unlock_keychain(kc_passwd)
+      [@security, 'unlock-keychain', '-p', kc_passwd]
+    end
+
+    def add_certificates
+      if @keychain == ''
+        [@security, 'add-certificates', @cert]
+      else
+        [@security, 'add-certificates', @cert, '-k', @keychain]
+      end
+    end
+
+    def import(cert_passwd)
+      [@security, 'import', @cert, '-P', cert_passwd, @keychain]
+    end
+
+    def install_certificate(cert_passwd)
+      if ::File.extname(@cert) == '.p12'
```

Comment:
In the [X.509 standard](https://en.wikipedia.org/wiki/X.509#Certificate_filename_extensions), there are a few different file extensions used for the various types of files. For certificates, we might find them with `.der`, `.crt`, or `.cer` and for [ PKCS #12](https://en.wikipedia.org/wiki/PKCS_12) files, we might find them with `.p12` of `pfx`. 

Might it make sense to check for these possible variations? Could we test for that?

Reply:
The 'import' command can actually support the formats: openssl, bsafe, raw, pkcs7, pkcs8, pkcs12, x509, openssh1, openssh2, and pemseq. We can probably make an array of acceptable file formats and check against those.